test/unit/route-reading.js

```diff
@@ -0,0 +1,85 @@
+/* global afterEach, before, beforeEach, describe, document, expect, inject, it, module, spyOn */
+'use strict';
+
+describe('Reading from $route service', function() {
+  beforeEach(module('angular-google-analytics'));
+  beforeEach(module(function (AnalyticsProvider) {
+    AnalyticsProvider
+      .setAccount('UA-XXXXXX-xx')
+      .logAllCalls(true)
+      .enterTestMode();
+  }));
+  
+  afterEach(inject(function (Analytics) {
+    Analytics.log.length = 0; // clear log
+  }));
+  
+  it('should not activate $route reading', function() {
+    inject(function(Analytics) {
+      expect(Analytics.configuration.readFromRoute).toBe(false);
+    });
+  });
+  
+  describe('without $route service available', function() {
+    beforeEach(module(function(AnalyticsProvider, $provide){
+      AnalyticsProvider.readFromRoute(true);
+      $provide.service('$log', function() {
```

Comment:
You should use a sinon spy for this instead of writing your own custom spy. This is done in a couple of other test functions. Here is a snippet:

```javascript
it('should issue a warning to the log', function () {
        inject(function ($log) {
          spyOn($log, 'warn');
          inject(function (Analytics) {
            expect(Analytics.log.length).toBe(1);
            expect(Analytics.log[0]).toEqual(['warn', 'No account id set to create analytics script tag']);
            expect($log.warn).toHaveBeenCalledWith(['No account id set to create analytics script tag']);
          });
        });
      });
```